src/elements/element.line.js

```diff
@@ -232,40 +423,121 @@ class Line extends Element {
 		}
 		const options = me.options;
 		if (options.tension && !options.steppedLine) {
-			updateBezierControlPoints(me._children, options, chartArea, me._loop);
+			const loop = options.spanGaps ? me._loop : me._fullLoop;
+			updateBezierControlPoints(me._points, options, chartArea, loop);
 		}
 	}
 
-	drawPath(ctx, area) {
+	set points(points) {
+		this._points = points;
+		delete this._segments;
+	}
+
+	get points() {
+		return this._points;
+	}
+
+	get segments() {
+		return this._segments || (this._segments = computeSegments(this));
+	}
+
+	/**
+	 * First non-skipped point on this line
+	 * @returns {Point|undefined}
+	 */
+	first() {
+		const segments = this.segments;
+		const points = this.points;
+		return segments.length && points[segments[0].start];
+	}
+
+	/**
+	 * Last non-skipped point on this line
+	 * @returns {Point|undefined}
+	 */
+	last() {
+		const segments = this.segments;
+		const points = this.points;
+		const count = segments.length;
+		return count && points[segments[count - 1].end];
+	}
+
+	/**
+	 * Interpolate a point in this line at the same value on `property` as
+	 * the reference `point` provided
+	 * @param {Point} point - the reference point
+	 * @param {string} property - the property to match on
+	 * @returns {Point|undefined}
+	 */
+	interpolate(point, property) {
 		const me = this;
 		const options = me.options;
-		const spanGaps = options.spanGaps;
-		let closePath = me._loop;
-		let points = me._children;
+		const value = point[property];
+		const points = me.points;
+		const segments = _boundSegments(me, {property, start: value, end: value});
 
-		if (!points.length) {
+		if (!segments.length) {
 			return;
 		}
 
-		if (closePath) {
-			points = startAtGap(points, spanGaps);
-			closePath = points.closePath;
+		const result = [];
+		const _interpolate = getInterpolationMethod(options);
+		let i, ilen;
+		for (i = 0, ilen = segments.length; i < ilen; ++i) {
+			const {start, end} = segments[i];
+			const p1 = points[start];
+			const p2 = points[end];
+			if (p1 === p2) {
+				result.push(p1);
+				continue;
+			}
+			const t = Math.abs((value - p1[property]) / (p2[property] - p1[property]));
+			let interpolated = _interpolate(p1, p2, t, options.steppedLine);
+			interpolated[property] = point[property];
+			result.push(interpolated);
+		}
+		return result.lenght === 1 ? result[0] : result;
+	}
+
+	/**
+	 * Append a segment of this line to current path.
+	 * @param {CanvasRenderingContext2D} ctx
+	 * @param {object} segment
+	 * @param {object} options
+	 */
+	pathSegment(ctx, segment, options) {
+		return pathSegment(ctx, this, segment, options);
+	}
+
+	/**
+	 * Append line to current path.
+	 * @param {CanvasRenderingContext2D} ctx
+	 * @param {object} params
```

Comment:
it'd be useful to fill this out. I'm not quite sure what it might contain or where it comes from. the only place I see this method being called only `ctx` is passed in. i'm probably missing something, but just wanted to double check that it's necessary